tests/source/fields.rs

```diff
@@ -0,0 +1,25 @@
+#![crate_type = "lib"]
+
+// @has /data/relationships/structs/data/0/type 'struct'
+// @has /data/relationships/structs/data/0/id 'fields::StructWithFields'
+
+// @has /included/0/type 'struct'
+// @has /included/0/id 'fields::StructWithFields'
+// @has /included/0/attributes/name 'StructWithFields'
+// @has /included/0/attributes/docs ''
+
+// This will fail, the type of 'fields::StructWithFields::integer' is 'struct'
+// has /included/0/relationships/child_structs/data/0/type 'field'
```

Comment:
I expect this to be a `field`, but it's a `struct`. It seems "children" of something all get set to have the type as the parent, which of course is not true for complex types like Structs or Enums.

Reply:
Hm, that seems... less than ideal.